File: LeetCode/is-subsequence.cpp
// https://leetcode.com/problems/is-subsequence/
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0;
        for(const auto& ch: t) {
            if(ch == s[i]) {
                i++;
            }
        }
        return i == s.size();
    }
};

File: LeetCode/is-subsequence.java
// https://leetcode.com/problems/is-subsequence/

class Solution {
  public boolean isSubsequence(String substring, String t) {
    if (substring.length() == 0) {
      return true;
    }

    int i = 0;
    for (char ch : t.toCharArray()) {
      if (ch == substring.charAt(i)) {
        i++;

        if (i == substring.length()) {
          return true;
        }
      }
    }

    return false;
  }
}